src/blockchain/transactions.rs

```diff
@@ -47,12 +49,44 @@ impl SignInput {
     }
 }
 
+impl TransactionConfirmations {
+    /// Adds confirmation from the specified anchoring node.
+    fn confirm_by_node(&mut self, anchoring_node_id: u16, public_key: btc::PublicKey) {
+        self.0.insert(anchoring_node_id, public_key);
+    }
+
+    /// Checks if there are enough confirmations to mark transaction as funding.
+    pub(crate) fn has_enough_confirmations(&self, config: &Config) -> Result<bool, ExecutionError> {
+        let confirmations = self
+            .0
+            .iter()
+            .try_fold(0, |count, (anchoring_node_id, public_key)| {
+                let expected_public_key = config
+                    .anchoring_keys
+                    .get(*anchoring_node_id as usize)
+                    .map(|x| &x.bitcoin_key);
+                if expected_public_key != Some(public_key) {
+                    Err(Error::MalformedFundingTxConfirmation)
+                } else {
+                    Ok(count + 1)
+                }
+            });
+        let quorum = exonum::helpers::byzantine_quorum(config.anchoring_keys.len());
```

Comment:
Maybe it makes sense to add a method `byzantine_quorum` to the `impl Config`, so we will be able to simply call `config.byzantine_quorum()`?